pkg/daemon/on_disk_validation.go

```diff
@@ -101,7 +103,17 @@ func checkV3Unit(unit ign3types.Unit, systemdPath string) error {
 		return nil
 	}
 
-	return checkFileContentsAndMode(path, []byte(*unit.Contents), defaultFilePermissions)
+	err := checkFileContentsAndMode(path, []byte(*unit.Contents), defaultFilePermissions)
+	if err != nil {
+		return err
+	}
+
+	err = checkUnitEnabled(unit.Name, unit.Enabled)
+	if err != nil {
+		return err
+	}
+
+	return err
```

Comment:
I think we can just condense these lines into `return checkUnitEnabled(unit.Name, unit.Enabled)` since we're returning `err` regardless, or make this `return nil` if you want to be explicit

Reply:
updated.